src/Symfony/Component/DependencyInjection/Dumper/Preloader.php

```diff
@@ -76,6 +77,10 @@ private static function doPreload(string $class, array &$preloaded): void
         $preloaded[$class] = true;
 
         try {
+            if (!class_exists($class)) {
```

Comment:
shouldn't this be `if (!class_exists($class) && !interface_exists($class, false) && !trait_exists($class, false)) {`?

but why in the first place, since there is a try/catch that should do the same?

Reply:
You are right, I'll update the PR.

this code is faster and relying on try/catch => generating and exception is costly in PHP. here it's a simple if statement

Reply:
But unless you have numbers about this, I don't think this is worth optimizing :)